config/v1/types_infrastructure.go

```diff
@@ -505,9 +505,151 @@ type BareMetalPlatformStatus struct {
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
 }
 
+type OpenStackAPIBGPPeer struct {
+	// asn is the Autonomous System number of the peer.
```

Comment:
The ASN is a 32-bit unsigned integer ([RFC 4271](https://www.rfc-editor.org/rfc/rfc4271), [RFC 6793](https://www.rfc-editor.org/rfc/rfc6793)) that identifies the autonomous system that pre-exists in the target OpenStack cluster.

The user knows this information from the moment they decide to use BGP in their infrastructure (as in: well before they install OCP).

Might be that I'm too much into the topic, but I can't figure out what information a user might be missing here, that they are not supposed to already possess when dealing with BGP. Do you have any suggestion to make this godoc better?

Reply:
actually, @mdbooth why did we choose `string` instead of `int` here?

Reply:
Normally when there's information from a system external to openshift, we document where they might find that information if they don't have it or where we would expect it to have come from. So, this will have been generated when you set up xyz may be appropriate here?